server/middleware/roleBasedAccessControl.ts

```diff
@@ -1,15 +1,76 @@
 import authorisationMiddleware from './authorisationMiddleware'
+import ApplicationRole from '../enums/applicationRole'
+import ApplicationAction from '../enums/applicationAction'
 
-enum ApplicationRoles {
-  ROLE_EDUCATION_WORK_PLAN_EDITOR = 'ROLE_EDUCATION_WORK_PLAN_EDITOR',
-}
+/**
+ * Deprecated - use `checkUserHasPermissionTo` instead
+ */
+const checkUserHasEditAuthority = () => authorisationMiddleware([ApplicationRole.ROLE_EDUCATION_WORK_PLAN_EDITOR])
 
-enum DpsRoles {
-  ROLE_GLOBAL_SEARCH = 'ROLE_GLOBAL_SEARCH',
-  ROLE_INACTIVE_BOOKINGS = 'ROLE_INACTIVE_BOOKINGS',
-  ROLE_POM = 'ROLE_POM',
+/**
+ * A map of [ApplicationAction] to [ApplicationRole]s, to determine which role is required for any given action.
+ * The list of [ApplicationRole]s should be considered an "or" list. Users require any one of the listed roles for each
+ * action.
+ */
+const rolesForAction = {
```

Comment:
This is the crux of how it all works ... a map of actions to the roles required to perform that action.

At the moment I've added all 3 of our roles to every action, which will mean stuff will continue to work whilst we get the roles created and migrate users to the new roles.
Once we've done that we can remove `ROLE_EDUCATION_WORK_PLAN_EDITOR`, and set the right role(s) to each action 👍